### PR TITLE
Feat/#175 homebrew release

### PR DESCRIPTION
**Pull request by sFritsch09**
I needed to edit the build pipeline because using the build steps for gorelease you need to have the Pro version: https://goreleaser.com/customization/builds/#import-pre-built-binaries

Anyway I kept the same binary structure as before and the pipeline is much shorter with goreleaser now.

✅  Added Homebrew with GoReleaser
✅  Edit build pipeline and tested workflow

In order to let GoRelease push the artefacts you need to add GITHUB_TOKEN secret:
**The minimum permissions the GITHUB_TOKEN should have to run this are write:packages**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -114,62 +114,11 @@ jobs:
...
-        path: pgroll.macos.amd64
-
-    - uses: actions/upload-artifact@v3
-      with:
-        name: pgroll.macos.arm64
-        path: pgroll.macos.arm64
-
-    - uses: actions/upload-artifact@v3
+    - name: Run GoReleaser
+      uses: goreleaser/goreleaser-action@v3
```
Comment (andrew-farries) [inline .github/workflows/build.yml:118]: The latest version of the action is `v5` [[here](https://github.com/goreleaser/goreleaser-action)]. Is there a particular reason why we don't use the latest?

File: .goreleaser.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      - darwin
+    goarch:
+      - amd64
+      - arm64
+    ignore:
+      - goos: windows
+        goarch: arm64
+
+brews:
+  -
```
Comment (andrew-farries) [inline .goreleaser.yaml:26]: don't need the extra newline here.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -114,62 +114,11 @@ jobs:
...
-          artifacts/pgroll.linux.amd64/pgroll.linux.amd64
-          artifacts/pgroll.linux.arm64/pgroll.linux.arm64
-          artifacts/pgroll.macos.amd64/pgroll.macos.amd64
-          artifacts/pgroll.macos.arm64/pgroll.macos.arm64
-          artifacts/pgroll.win.amd64/pgroll.win.amd64
+        distribution: goreleaser
+        version: latest
+        args: release --clean
+      env:
+        GITHUB_TOKEN: ${{ secrets.GH_TOKEN }}
```
Comment (andrew-farries) [inline .github/workflows/build.yml:124]: This is already in the repository as `GIT_TOKEN`:

```suggestion
        GITHUB_TOKEN: ${{ secrets.GIT_TOKEN }}
```

**Review by andrew-farries — CHANGES_REQUESTED** [review]
Thanks for the contribution @sFritsch09 👍 

I've made some suggestions.

- [x] I'm going to create the `homebrew-pgroll` repository. [[repo](https://github.com/xataio/homebrew-pgroll/)])

File: .github/workflows/build.yml (hunk 1 above)
Comment (sFritsch09) [reply .github/workflows/build.yml:118]: will use v5 now

File: .goreleaser.yaml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    name: pgroll
+    homepage: "https://www.xata.io"
+    description: "Postgres zero-downtime migrations made easy"
+    repository:
+      owner: xataio
+      name: pgroll
```
[the remark this answers is not in the document]
Comment (sFritsch09) [reply .goreleaser.yaml:32]: Right, I also saw this in the documentation

File: .goreleaser.yaml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+      - windows
+      - darwin
+    goarch:
+      - amd64
+      - arm64
+    ignore:
+      - goos: windows
+        goarch: arm64
+
+brews:
```
[the remark this answers is not in the document]
Comment (sFritsch09) [reply .goreleaser.yaml:24]: I added homebrew install construction to the docs

**Comment by sFritsch09** [comment]
Pleasure @andrew-farries, it's my first PR so far, appreciate your input!
I guess everything is in order following the implementation of your suggestions. Please inform me if there are any remaining matters to address :)

**Review by andrew-farries — CHANGES_REQUESTED** [review]

File: README.md [hunk 6]
```diff
@@ -159,8 +169,8 @@ For more advanced usage, a tutorial, and detailed options refer to the full [Doc
 
 We welcome contributions from the community! If you'd like to contribute to `pgroll`, please follow these guidelines:
 
-* Create an [issue](https://github.com/xataio/pgroll/issues) for any questions, bug reports, or feature requests.
-* Check the documentation and [existing issues](https://github.com/xataio/pgroll/issues) before opening a new issue.
+- Create an [issue](https://github.com/xataio/pgroll/issues) for any questions, bug reports, or feature requests.
+- Check the documentation and [existing issues](https://github.com/xataio/pgroll/issues) before opening a new issue.
```
Comment (andrew-farries) [inline README.md:173]: This block of changes can be reverted.

File: README.md [hunk 7]
```diff
@@ -176,11 +186,11 @@ For this project, we pledge to act and interact in ways that contribute to an op
...
-* [Reshape](https://github.com/fabianlindfors/reshape) by Fabian Lindfors
-* [PgHaMigrations](https://github.com/braintree/pg_ha_migrations)
-* [PostgreSQL at Scale: Database Schema Changes Without Downtime](https://medium.com/paypal-tech/postgresql-at-scale-database-schema-changes-without-downtime-20d3749ed680)
-* [Zero downtime schema migrations in highly available databases](http://essay.utwente.nl/92098/1/vanKampen_MA_EEMCS.pdf)
-* [Expand and contract pattern](https://openpracticelibrary.com/practice/expand-and-contract-pattern/)
+- [Reshape](https://github.com/fabianlindfors/reshape) by Fabian Lindfors
+- [PgHaMigrations](https://github.com/braintree/pg_ha_migrations)
+- [PostgreSQL at Scale: Database Schema Changes Without Downtime](https://medium.com/paypal-tech/postgresql-at-scale-database-schema-changes-without-downtime-20d3749ed680)
+- [Zero downtime schema migrations in highly available databases](http://essay.utwente.nl/92098/1/vanKampen_MA_EEMCS.pdf)
+- [Expand and contract pattern](https://openpracticelibrary.com/practice/expand-and-contract-pattern/)
```
Comment (andrew-farries) [inline README.md:193]: This block of changes can be reverted.

File: docs/README.md [hunk 8]
```diff
@@ -100,6 +100,16 @@ go install github.com/xataio/pgroll@latest
...
+### From package manager - Homebrew
+
+To install `pgroll` with homebrew, run the following command:
+
+```sh
+# macOS or Linux
+brew tap xataio/pgroll
+brew install pgroll
+```
+
```
Comment (andrew-farries) [inline docs/README.md:112]: Same comment in this file about reverting the other changes and leaving only these new installation instructions.

File: README.md [hunk 9]
```diff
@@ -90,33 +99,34 @@ Create a migration file. You can check the [examples](examples) folder for some
...
+					},
+					{
+						"name": "bio",
+						"type": "text",
+						"nullable": true
+					}
+				]
+			}
+		}
+	]
```
Comment (andrew-farries) [inline README.md:126]: this indentation change can be reverted.

**Review by andrew-farries — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml [hunk 10]
```diff
@@ -101,9 +101,9 @@ jobs:
           fi
         done
 
-  build:
+  release:
     runs-on: ubuntu-latest
-    needs: [test, lint, examples]
+    needs: [test, lint, examples, license-check]
```
Comment (andrew-farries) [inline .github/workflows/build.yml:106]: We also need to restore the `if: startsWith(github.ref, 'refs/tags/')` to ensure that the step is only triggered on a push of a `semver` tag.

**Review by andrew-farries — CHANGES_REQUESTED** [review]

**Comment by philkra** [comment]
Hi @sFritsch09 , thank you very much for your contribution! We would love to send you some swag, could you reach out to me via email: philip@xata.io to coordinate? Cheers, Philip

File: docs/README.md (hunk 8 above)
Comment (sFritsch09) [reply docs/README.md:112]: Alright I needed to save without formatting in VSCode

File: README.md (hunk 9 above)
Comment (sFritsch09) [reply README.md:126]: saved without formatting

File: .github/workflows/build.yml (hunk 10 above)
Comment (sFritsch09) [reply .github/workflows/build.yml:106]: added tag condition to release job

**Review by andrew-farries — APPROVED** [review]
I've removed the remaining formatting changes from `docs/README.md` and updated the repo's `settings.yaml` file.

This looks good to go now 🚢 

Thanks for your contribution @sFritsch09! ❤️

**Pull request merged**